lib/src/wrapWithVinylFile.js

```diff
@@ -16,7 +16,18 @@ function resolveSymlinks(options) {
       globFile.stat = stat;
 
       if (!stat.isSymbolicLink() || !options.followSymlinks) {
-        return cb(null, globFile);
+        var vinylFile = new File(globFile);
+        if (globFile.symlinkTarget) {
+          // If we reach here, it means there is at least one
+          // symlink on the path and we need to rewrite the path
+          // to its original value.
+          // Updated file stats will tell getContents() to actually read it.
+          vinylFile.path = globFile.symlinkTarget;
+        }
+        return cb(null, vinylFile);
+      }
+      if (!globFile.symlinkTarget) {
```

Comment:
I think this should be inside the `fs.realpath` callback before we modify the `globFile.path`